pkg/sql/schemachanger/scplan/internal/rules/dep_index_and_column.go

```diff
@@ -85,22 +85,65 @@ func init() {
 					(*scpb.IndexName)(nil),
 					(*scpb.IndexPartitioning)(nil),
 					(*scpb.IndexComment)(nil),
+					(*scpb.IndexColumn)(nil),
 				),
 				joinOnIndexID(from.el, to.el, "table-id", "index-id"),
 				targetStatusEq(from.target, to.target, scpb.ToPublic),
 				currentStatus(from.node, scpb.Status_BACKFILL_ONLY),
 				currentStatus(to.node, scpb.Status_PUBLIC),
 			}
 		})
+	registerDepRule(
+		"index existence precedes index dependents",
+		scgraph.SameStagePrecedence,
+		"index", "index-column",
+		func(from, to nodeVars) rel.Clauses {
+			return rel.Clauses{
+				from.el.Type(
+					(*scpb.PrimaryIndex)(nil),
+					(*scpb.SecondaryIndex)(nil),
+				),
+				to.el.Type(
+					(*scpb.IndexColumn)(nil),
+				),
+				joinOnIndexID(from.el, to.el, "table-id", "index-id"),
+				targetStatusEq(from.target, to.target, scpb.ToPublic),
+				currentStatus(from.node, scpb.Status_BACKFILL_ONLY),
+				currentStatus(to.node, scpb.Status_PUBLIC),
+			}
+		})
+	registerDepRule(
+		"temp index existence precedes index dependents",
+		scgraph.SameStagePrecedence,
+		"index", "index-column",
+		func(from, to nodeVars) rel.Clauses {
+			return rel.Clauses{
+				from.el.Type(
+					(*scpb.PrimaryIndex)(nil),
+					(*scpb.SecondaryIndex)(nil),
```

Comment:
The rule name says "*temp* index existence preceded index dependents" but why isn't the from type "scpb.TemporaryIndex"?

Reply:
Good catch :)